### DISCUSSION TITLE
EBNF format input transform to LARK format

### DISCUSSION THREAD

**Discussion by Fxztam** [lark-parser/lark#848] [Q&A]
Hello,

I created the EBNF of a query language completely in the format e.g.:

    _OrderByClause = "ORDER" "BY" OrderTerm [ { "," OrderTerm } ];_

this EBNF format works fine with some editors and verifying tools as `quasi default`.

Is there a way to transform this syntax to LARK's definition syntax? 

Thanks,
Friedhold

**Comment by MegaIng** [comment]
There is not build in way of doing this at the moment, but it is somewhat trivial to write your own Lark grammar to match your EBNF syntax and then convert to lalr.

**Comment by ThatXliner** [comment]
Turn EBNF into Lark (using Lark, ofc)? Sounds like a fun idea! <sub>now I just need the grammar specification for the grammar specification 😆</sub>
